File: lang-card-backend/models/flashcard.py
from pydantic import BaseModel
from typing import List, Optional, Literal


class Flashcard(BaseModel):
    term_native: str
    term_learning_language: str
    example_sentence_learning_language: str
    term_audio: Optional[str]
    sentence_audio: Optional[str]
    active: bool


class Flashcards(BaseModel):
    flashcards: List[Flashcard]


FlashcardContentType = Literal[
    "term_learning_language", "term_native", "example_sentence_learning_language"
]


class FlashcardFormat(BaseModel):
    front: List[FlashcardContentType]
    back: List[FlashcardContentType]

File: lang-card-frontend-v2/src/app/constants/Flashcard.ts
type Flashcard = {
  term_native: string;
  term_learning_language: string;
  example_sentence_learning_language: string;
  term_audio?: string;
  sentence_audio?: string;
  active: boolean;
};

type FlashcardSide =
  | "term_native"
  | "term_learning_language"
  | "example_sentence_learning_language";

type FlashcardFormat = {
  front: FlashcardSide[];
  back: FlashcardSide[];
};

export type { Flashcard, FlashcardSide, FlashcardFormat };
